src/PostgREST/Error.hs

```diff
@@ -99,11 +99,19 @@ instance JSON.ToJSON ApiRequestError where
     "message" .= ("More than one relationship was found for " <> parent <> " and " <> child :: Text),
     "details" .= (compressedRel <$> rels) ]
   toJSON (AmbiguousRpc procs)  = JSON.object [
-    "hint"    .= ("Overloaded functions with the same argument name but different types are not supported" :: Text),
-    "message" .= ("Could not choose the best candidate function between: " <> T.intercalate ", " [pdSchema p <> "." <> pdName p <> "(" <> T.intercalate ", " [pgaName a <> " => " <> pgaType a | a <- pdArgs p] <> ")" | p <- procs])]
-  toJSON (NoRpc schema procName payloadKeys hasPreferSingleObject)  = JSON.object [
-    "hint"    .= ("If a new function was created in the database with this name and arguments, try reloading the schema cache." :: Text),
-    "message" .= ("Could not find the " <> schema <> "." <> procName <> (if hasPreferSingleObject then " function with a single json or jsonb argument" else "(" <> T.intercalate ", " payloadKeys <> ")" <> " function") <> " in the schema cache")]
+    "hint"    .= ("Try changing the parameters' names or the function name so function overloading can be resolved" :: Text),
+    "message" .= ("Could not choose the best candidate function between: " <> T.intercalate ", " [pdSchema p <> "." <> pdName p <> "(" <> T.intercalate ", " [ppName a <> " => " <> ppType a | a <- pdParams p] <> ")" | p <- procs])]
```

Comment:
> I think it should be specified that the change should be done in the database, otherwise, at first glance, it may look like the change should be done on the query

True, I agree.

> Also, I think the old hint should be kept and moved to the "detail" field. WDYT?

The thing is that the "Overloaded functions with the same argument name but different types are not supported" error is no longer accurate now. This can be noted on the test I've added:

```haskell
it "will not be able to resolve when a single unnamed json parameter exists and other overloaded functions exist" $
  request methodPost "/rpc/overloaded_unnamed_param" [("Content-Type", "application/json")]
      [json|{"x": 1, "y": 2}|]
    `shouldRespondWith`
      [json| {
        "hint":"Try changing the parameters' names or the function name so function overloading can be resolved",
        "message":"Could not choose the best candidate function between: test.overloaded_unnamed_param( => json), test.overloaded_unnamed_param(x => integer, y => integer)"}|]
      { matchStatus  = 300
      , matchHeaders = [matchContentTypeJson]
      }
```

The above is not related to having the same parameter name with different types.